### DISCUSSION TITLE
Add support of GenAI to rapidly assist in content creation

### DISCUSSION THREAD

**Discussion by SharadKumar** [Thinkmill/keystatic#1119] [Ideas]
It would be super useful if we could have context provided in config (or also on admin) and have a GenAI sparkle icon next to text input/textarea fields in Keystatic admin. Admins setup their own OpenAI compatible api keys of course. 

For example:
https://dub.co/blog/introducing-dub-ai

Vercel SDK released yesterday makes it much easier (cleaner) to implement. Also you get compatibility with various models.
https://sdk.vercel.ai/docs/ai-sdk-core/generating-structured-data

This shall bring utility of Keystatic admin and content authoring to another level.

**Comment by simonswiss** [comment]
Ummmm, so... 🤗

https://twitter.com/thekeystatic/status/1788462053038125520

**Reply by dmitryrechkin** [reply]
@simonswiss Have there been any updates on this feature? If you’re short on time, I’d be happy to tackle it myself - I just need a few pointers on where to start or which file to modify. Thanks so much, and I love your product!

**Reply by dmitryrechkin** [reply]
@simonswiss sorry for bugging you, I assume you may not have time to look into adding AI into keystatic, so I thought to ask if you can point me in the right direction. I want to understand if there is any easy way to get selected content in the editor? 
I want to see if it is possible to grab selected text and modify it programatically.

Thanks in advance
